src/MiniCover/Reports/coverallsReport.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.Collections.Generic;
+using MiniCover.Model;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Security.Cryptography;
+using System.Net.Http;
+using Newtonsoft.Json;
+
+namespace MiniCover.Reports
+{
+
+    class CoverallLine
+    {
+        public int InstructionId = -1;
+        public int InstructionLineCounts = int.MaxValue;
+        public int Hits;
+    }
+
+    public class CoverallsReport : BaseReport
+    {
+
+        private const string BgColorGreen = "background-color: #D2EACE;";
+        private const string BgColorRed = "background-color: #EACECC;";
+        private const string BgColorBlue = "background-color: #EEF4ED;";
+
+        private readonly string _output;
+        private readonly string _job_id;
+        private readonly string _repo_token;
+        private readonly string _service_name;
+        private readonly string _message;
+        private readonly string _rootfolder;
+        private readonly string _post;
+        private readonly string _git_id;
+        private readonly string _git_branch;
+        private readonly string _git_remote_name;
+        private readonly string _git_remote_url;
+        //private readonly float  _threshold;
+
+        public CoverallsReport(string output
+            , string repo_token
+            , string job_id
+            , string service_name
+            , string message
+            , string rootfolder
+            , string post
+            , string git_id
+            , string git_branch
+            , string git_remote_name
+            , string git_remote_url
+            //, float threshold
+            )
+        {
+            _output = output;
+            _repo_token = repo_token;
+            _job_id = job_id;
+            _service_name = (service_name ?? "local");
+            _rootfolder = rootfolder;
+            _post = post;
+            _git_id = git_id;
+            _git_branch = (git_branch ?? "master");
+            _git_remote_name = git_remote_name;
+            _git_remote_url = git_remote_url;
+            _message = (message ?? "--");
+            //_threshold = threshold;
+
+            if(null == _repo_token)
+                throw new ArgumentException("token not specified");
```

Comment:
It seems that the token is not mandatory:
https://github.com/lemurheavy/coveralls-public/issues/208